### PR TITLE
Feature: add force and stress output for all esolver

### LINKED ISSUE #3060: output force and stress with dp potential in relax calculation
### Background

Dear all,

**abacus v3.4.0 with deepmd**

I want to do some `relax` and `cell-relax` with dp potential. I have set `cal_stress=1` and `cal_force=1` in the `INPUT` file,  However, the forces and stress were not ouput in the `log` file. When the `calculation` type is `MD`, the forces and stress would output.
This kind of issure was found when I used dpgen autotest, it needs the stress from log file. 

### Describe the solution you'd like

you can do it.

### Additional Context

_No response_

### Task list for Issue attackers

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

### PR DESCRIPTION
**Pull request by YuLiu98**
fix #3060

### CODE REVIEW

File: source/module_io/output_log.cpp [hunk 1]
```diff
@@ -27,4 +27,177 @@ void output_efermi(bool& convergence, double& efermi, std::ofstream& ofs_running
     }
 }
 
+void print_force(std::ofstream& ofs_running,
```
Comment (dyzheng) [inline source/module_io/output_log.cpp:32]: @kirk0830 Any comments for using FMT tools here?

Comment (kirk0830) [inline source/module_io/output_log.cpp:32]: Hi, I would suggest use the newly implemented class formatter to control the format output, for a quick example, please have a look at this function: https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_elecstate/elecstate_print.cpp#L154
. Formatter gets data column-by-column, you can find more flexible usage on Feishu manual: https://ucoyxk075n.feishu.cn/docx/Yym9dnm3aoTMfHxin8rcX9Rvnmb.
Besides elecstate_print, you can also find the use of formatter tools in timer.cpp: https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_base/timer.cpp#L230
and read_atoms.cpp: https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_cell/read_atoms.cpp#L935
Many thanks~

**Review by kirk0830 — COMMENTED** [review]
thanks for paying efforts on refactoring print_* functions with new formatter library!
I notice you have changed reference values of integrated tests, I would suggest add some additonal explanation on these changes,
thanks.

**Review by dyzheng — APPROVED** [review]
The uniform format of force and stress will make the program user friendly, good job!

**Pull request merged**